Update datasets library to 2.2.2 in requirements

VijayKalmath opened the pull request:
# What does this PR do?

## Summary
*Textattack currently installs `datasets 1.15` while installing textattack , which is far behind the latest datasets library.*

*This commit sets datasets to `2.2.2` in requirments.txt and updates test cases to suit the default imdb dataset used in `datasets 2.2.2`.*

## Additions


## Changes
- *changed datasets to 2.2.2 in `requirements.txt`.*
- *updated test case in `tests/sample_outputs/run_attack_from_file.txt` with the new desired output to stay updated with data used by  `datasets=2.2.2`.*

## Deletions

## Issues Addressed 
Fixes #590 

## Checklist
- [X] The title of your pull request should be a summary of its contribution.
- [X] Please write detailed description of what parts have been newly added and what parts have been modified. Please also explain why certain changes were made.
- [X] If your pull request addresses an issue, please mention the issue number in the pull request description to make sure they are linked (and people consulting the issue know you   are working on it)
- [X] To indicate a work in progress please mark it as a draft on Github.
- [X] Make sure existing tests pass.
- [  ] Add relevant tests. No quality testing = no merge.
- [  ] All public methods must have informative docstrings that work nicely with sphinx. For new modules/files, please add/modify the appropriate `.rst` file in `TextAttack/docs/apidoc`.'

jxmorris12 commented:
Sorry, but the three skipped examples feels a bit odd, doesn't it? For a model with >80% accuracy, that's pretty unlikely that it gets three wrong back-to-back.

jxmorris12 commented on tests/sample_outputs/run_attack_from_file.txt:
```diff
@@ -1,4 +1,4 @@
-/.*/Attack(
```
You need to add this /.*/ back in for the test to pass. 

It's the regex for "any character" and helps us match the text in Github Actions, since there's a bunch of extra stuff printed out when this attack is run on the VM for the first time.

VijayKalmath replied:
Oh interesting. The tests were successful on my VM so I went ahead with it .

VijayKalmath commented:
Adding Model Evaluations of cnn-imdb and lstm-imdb on imdb datasets across the 2 versions of the datasets package. 

Datasets - 1.15 ;  huggingface-hub-0.0.19
-- 

```
(Env_TextAttack) ecbm4040@instance-2:~/TextAttack$ textattack eval --model cnn-imdb --num-examples -1
textattack: Loading pre-trained TextAttack CNN: cnn-imdb
Reusing dataset imdb (/home/ecbm4040/.cache/huggingface/datasets/imdb/plain_text/1.0.0/e3c66f1788a67a89c7058d97ff62b6c30531e05b549de56d3ab91891f0561f9a)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 977.85it/s]
textattack: Loading datasets dataset imdb, split test.
textattack: Got 25000 predictions.
textattack: Correct 20346/25000 (81.38%)



(Env_TextAttack) ecbm4040@instance-2:~/TextAttack$ textattack eval --model cnn-imdb --num-examples -1 --dataset-split train
textattack: Loading pre-trained TextAttack CNN: cnn-imdb
Reusing dataset imdb (/home/ecbm4040/.cache/huggingface/datasets/imdb/plain_text/1.0.0/e3c66f1788a67a89c7058d97ff62b6c30531e05b549de56d3ab91891f0561f9a)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 951.81it/s]
textattack: Loading datasets dataset imdb, split train.
textattack: Got 25000 predictions.
textattack: Correct 21602/25000 (86.41%)


(Env_TextAttack) ecbm4040@instance-2:~/TextAttack$ textattack eval --model lstm-imdb --num-examples -1 --dataset-split train
textattack: Loading pre-trained TextAttack LSTM: lstm-imdb
Reusing dataset imdb (/home/ecbm4040/.cache/huggingface/datasets/imdb/plain_text/1.0.0/e3c66f1788a67a89c7058d97ff62b6c30531e05b549de56d3ab91891f0561f9a)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 889.31it/s]
textattack: Loading datasets dataset imdb, split train.
textattack: Got 25000 predictions.
textattack: Correct 21269/25000 (85.08%)
```

Datasets - 2.2.2; huggingface-hub-0.7.0
--
```
(Env_TextAttack) ecbm4040@instance-2:~/TextAttack$ textattack eval --model cnn-imdb --num-examples -1
textattack: Loading pre-trained TextAttack CNN: cnn-imdb
Reusing dataset imdb (/home/ecbm4040/.cache/huggingface/datasets/imdb/plain_text/1.0.0/2fdd8b9bcadd6e7055e742a706876ba43f19faee861df134affd7a3f60fc38a1)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 891.27it/s]
textattack: Loading datasets dataset imdb, split test.
textattack: Got 25000 predictions.
textattack: Correct 20346/25000 (81.38%)


(Env_TextAttack) ecbm4040@instance-2:~/TextAttack$ textattack eval --model cnn-imdb --num-examples -1 --dataset-split train
textattack: Loading pre-trained TextAttack CNN: cnn-imdb
Reusing dataset imdb (/home/ecbm4040/.cache/huggingface/datasets/imdb/plain_text/1.0.0/2fdd8b9bcadd6e7055e742a706876ba43f19faee861df134affd7a3f60fc38a1)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 881.28it/s]
textattack: Loading datasets dataset imdb, split train.
textattack: Got 25000 predictions.
textattack: Correct 21602/25000 (86.41%)



(Env_TextAttack) ecbm4040@instance-2:~/TextAttack$ textattack eval --model lstm-imdb --num-examples -1 --dataset-split train
textattack: Loading pre-trained TextAttack LSTM: lstm-imdb
Reusing dataset imdb (/home/ecbm4040/.cache/huggingface/datasets/imdb/plain_text/1.0.0/2fdd8b9bcadd6e7055e742a706876ba43f19faee861df134affd7a3f60fc38a1)
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 888.94it/s]
textattack: Loading datasets dataset imdb, split train.
textattack: Got 25000 predictions.
textattack: Correct 21269/25000 (85.08%)
```

jxmorris12 commented:
hope the tests can pass now!

jxmorris12 commented:
LGTM! Thanks for doing this, we needed it.
